pkg/serverless/daemon/routes.go

```diff
@@ -86,6 +91,9 @@ type EndInvocation struct {
 
 func (e *EndInvocation) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	log.Debug("Hit on the serverless.EndInvocation route.")
+	e.daemon.ExecutionSpanStateLock.Lock()
+	defer e.daemon.ExecutionSpanStateLock.Unlock()
+	e.daemon.hitOnEnd = true
```

Comment:
You need to include tests for this. Write a test that shows that when `ServeHTTP` method is called, that the `hitOnEnd` is set to `true`. Same for the above endpoints.